lib/engine/game/base.rb

```diff
@@ -1377,12 +1377,12 @@ def compute_stops(route, train = nil)
           stops, revenue = visits.combination(num_stops.to_i).map do |stops|
             # Make sure this set of stops is legal
             # 1) At least one stop must have a token (if enabled)
-            next if train.requires_token && stops.none? { |stop| stop.tokened_by?(route.corporation) }
+            next if train.requires_token && stops.none? { |stop| stop&.tokened_by?(route.corporation) }
 
             # 2) We can't ask for more revenue centers of a type than are allowed
             types_used = Array.new(distance.size, 0) # how many slots of each row are filled
 
-            next unless stops.all? do |stop|
+            next unless stops.flatten.all? do |stop|
               row = distance.index.with_index do |h, i|
```

Comment:
When doing a route from Russia (A3) and Vladivostock (A15), the stops variable for some reason combines them into an array of two stops, and stop.type fails. This is the cause of the error reported in the bug.
Flatten obviously works, but i don't know why it combines it into an array

Reply:
See above comment